readthedocs/search/api.py

```diff
@@ -1,16 +1,53 @@
+import logging
+from pprint import pformat
+
 from rest_framework import generics
+from rest_framework import serializers
 from rest_framework.exceptions import ValidationError
+from rest_framework.pagination import PageNumberPagination
 
 from readthedocs.search.documents import PageDocument
-from readthedocs.search.filters import SearchFilterBackend
-from readthedocs.search.pagination import SearchPagination
-from readthedocs.search.serializers import PageSearchSerializer
 from readthedocs.search.utils import get_project_list_or_404
 
+log = logging.getLogger(__name__)
+
+
+class SearchPagination(PageNumberPagination):
+    page_size = 25
+    page_size_query_param = 'page_size'
+    max_page_size = 100
+
+
+class PageSearchSerializer(serializers.Serializer):
```

Comment:
Can we take the serializers to `serializers.py`?

Reply:
I found it a lot of overhead to keep moving around between files when they were only used in one place. I find it easier this way to have all the things we need for the API in one place.

Reply:
I also think we'll need to change our implementation here to support faceting. If we want to support facets, the current approach won't work, since the serializer never sees the facet objects.